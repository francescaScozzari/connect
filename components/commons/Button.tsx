import styled from 'styled-components'

export const Button = styled.button`
  align-items: center;
  background-color: white;
  border: 2px solid black;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  min-width: 120px;
  outline: none;
  padding: 1em 1.5em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  width: fit-content;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
`
