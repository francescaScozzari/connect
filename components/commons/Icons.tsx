import React from 'react'

type Props = {
  title: string
} & React.ComponentPropsWithoutRef<'svg'>

const IconArrowCircle = ({ title, ...props }: Props) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" {...props}>
      <title>{title}</title>
      <path d="M12 1a11 11 0 1 0 11 11A11 11 0 0 0 12 1Zm4.707 9.707a1 1 0 0 1-1.414 0L13 8.414V18a1 1 0 0 1-2 0V8.414l-2.293 2.293a1 1 0 1 1-1.414-1.414l4-4a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1 0 1.414Z" />
    </svg>
  )
}

const IconArrowSubmit = ({ title, ...props }: Props) => {
  return (
    <svg
      width="45"
      height="45"
      viewBox="0 0 45 45"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g id="Arrow forward">
        <path
          id="Vector"
          d="M22.5 7.5L19.8563 10.1437L30.3188 20.625H7.5V24.375H30.3188L19.8563 34.8563L22.5 37.5L37.5 22.5L22.5 7.5Z"
          fill="black"
        />
      </g>
    </svg>
  )
}

const IconCrossReset = ({ title, ...props }: Props) => {
  return (
    <svg
      width="31"
      height="31"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g>
        <path
          d="m24.5417 8.27962-1.8212-1.82125L15.5 13.6788 8.27962 6.45837 6.45837 8.27962 13.6788 15.5l-7.22043 7.2205 1.82125 1.8212L15.5 17.3213l7.2205 7.2204 1.8212-1.8212L17.3213 15.5l7.2204-7.22038Z"
          fill="#000"
        />
      </g>
    </svg>
  )
}

const IconCopy = ({ title, ...props }: Props) => {
  return (
    <svg
      width="49"
      height="49"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g>
        <path
          d="M32.6673 2.04163H8.16732c-2.24584 0-4.08334 1.8375-4.08334 4.08333V34.7083h4.08334V6.12496H32.6673V2.04163Zm-2.0416 8.16667H16.334c-2.2458 0-4.0629 1.8375-4.0629 4.0833l-.0204 28.5834c0 2.2458 1.817 4.0833 4.0629 4.0833h22.4787c2.2459 0 4.0834-1.8375 4.0834-4.0833V22.4583l-12.25-12.25ZM16.334 42.875V14.2916h12.25V24.5h10.2083v18.375H16.334Z"
          fill="#000"
        />
      </g>
    </svg>
  )
}

const IconTarget = ({ title, ...props }: Props) => {
  return (
    <svg
      width="49"
      height="51"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M24.4993 16.9969c-4.512 0-8.1666 3.803-8.1666 8.4984s3.6546 8.4985 8.1666 8.4985c4.5121 0 8.1667-3.8031 8.1667-8.4985 0-4.6954-3.6546-8.4984-8.1667-8.4984Zm18.2525 6.3738c-.9391-8.8596-7.697-15.89205-16.2108-16.86937V4.24924c0-1.16853-.9187-2.12461-2.0417-2.12461-1.1229 0-2.0416.95608-2.0416 2.12461v2.25209C13.9439 7.47865 7.18602 14.5111 6.24685 23.3707H4.08268c-1.12291 0-2.04166.9561-2.04166 2.1246 0 1.1686.91875 2.1246 2.04166 2.1246h2.16417c.93917 8.8597 7.69705 15.8921 16.21085 16.8694v2.2521c0 1.1686.9187 2.1246 2.0416 2.1246 1.123 0 2.0417-.956 2.0417-2.1246v-2.2521c8.5138-.9773 15.2717-8.0097 16.2108-16.8694h2.1642c1.1229 0 2.0417-.956 2.0417-2.1246 0-1.1685-.9188-2.1246-2.0417-2.1246h-2.1642ZM24.4993 40.3676c-7.9012 0-14.2916-6.65-14.2916-14.8723 0-8.2222 6.3904-14.8722 14.2916-14.8722 7.9013 0 14.2917 6.65 14.2917 14.8722 0 8.2223-6.3904 14.8723-14.2917 14.8723Z"
        fill="#000"
      />
    </svg>
  )
}

export {
  IconArrowCircle,
  IconArrowSubmit,
  IconCrossReset,
  IconCopy,
  IconTarget
}
