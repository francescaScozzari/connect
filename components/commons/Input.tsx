import React from 'react'
import styled from 'styled-components'

type Props = JSX.IntrinsicElements['input'] & {
  errors?: string[]
}

const InputWithErrors = ({
  errors,
  max,
  min,
  onChange,
  placeholder,
  required,
  step,
  style,
  type,
  value
}: Props) => {
  return (
    <InputContainer style={style}>
      <Input
        errors={errors && errors.length > 0}
        max={max}
        min={min}
        onChange={onChange}
        placeholder={placeholder}
        required={required}
        step={step}
        type={type}
        value={value}
      />
      {errors && errors.length > 0 && errors.map(err => <p key={err}>{err}</p>)}
    </InputContainer>
  )
}

const Input = styled.input<{ errors?: boolean }>`
  border: 2px solid black;
  box-sizing: border-box;
  border-radius: 0.5em;
  padding: 0.625em 2em;
  width: 25em;
  font-style: normal;
  font-size: 1.125rem;
  &::placeholder {
    padding-left: 1.5em;
    font-size: 1.125rem;
    background: url("data:image/svg+xml,%3Csvg height='24' width='24' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:none;stroke:%23000;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px%7D%3C/style%3E%3C/defs%3E%3Cg id='_21.search'%3E%3Ccircle class='cls-1' cx='9' cy='9' r='8'/%3E%3Cpath class='cls-1' d='m15 15 8 8'/%3E%3C/g%3E%3C/svg%3E")
      no-repeat 0.5em center;
  }
  @media (max-width: 768px) {
    font-size: 1em;
    width: 15em;
  }
`

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5em;
  p {
    font-size: 0.75em;
    margin-left: 0.625em;
    color: ${({ theme }) => theme.colors.status.error};
  }
`

export { Input, InputWithErrors }
