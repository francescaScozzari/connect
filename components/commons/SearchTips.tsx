import React from 'react'
import { styled } from 'styled-components'

import { Text } from '@/components/commons/Typography'
import { IconNote, IconTarget } from './Icons'

type Props = {
  titleColor?: string
  textColor?: string
}

const SearchTips = ({
  titleColor = '#2B2D42',
  textColor = '#242424'
}: Props) => {
  return (
    <Wrapper>
      <SubWrapper>
        <IconNote title={'note'} color={titleColor} />
        <Paragraph color={textColor}>
          <Title color={titleColor}>Start from the research call</Title>
          <br /> If you are working or will have to work on a research paper for
          a call, you might report in the search field the title or objectives
          of that
        </Paragraph>
      </SubWrapper>

      <SubWrapper>
        <IconTarget title={'target'} color={titleColor} />
        <Paragraph color={textColor}>
          <Title color={titleColor}>More detail for greater match</Title>
          <br /> Enter as much information, don't limit yourself to a single
          word, that way we can narrow it down and suggest the researchers best
          suited to your need
        </Paragraph>
      </SubWrapper>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  gap: 2em;
  max-width: 67em;
  margin: 4em 0;

  @media (max-width: 1280px) {
    max-width: 46em;
  }

  @media (max-width: 768px) {
    max-width: 37.5em;
  }
`

const SubWrapper = styled.div`
  display: flex;
  gap: 0.5em;
  padding: 1.5em;
  align-items: start;
  max-width: 29.85em;
  border: 1px solid #7a83ab;
  border-radius: 1.25em;

  svg {
    flex-shrink: 0;
  }
`

const Paragraph = styled(Text.Normal)<{ color: string }>`
  line-height: 1.5;
  color: ${({ color }) => color};
`

const Title = styled.span<{ color: string }>`
  display: inline-block;
  color: ${({ color }) => color};
  margin-bottom: 0.5em;
  font-weight: 600;
  line-height: 1.5;
`

export { SearchTips }
