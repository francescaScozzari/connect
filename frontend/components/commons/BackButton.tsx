import React from 'react'
import styled from 'styled-components'

import { IconSearch } from './Icons'
import { Text } from './Typography'

export const BackButton = () => {
  return (
    <Button>
      <IconSearch title="back" />
      <Label as="span">make a new search</Label>
    </Button>
  )
}

const Label = styled(Text.Normal)`
  text-transform: uppercase;
  font-weight: 600;
  line-height: normal;
  color: white;
`

const Button = styled.button`
  all: unset;
  display: flex;
  justify-content: center;
  align-content: center;
  border-radius: 10px;
  cursor: pointer;
  width: fit-content;
  outline: none;
  padding: 0.75em 1em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  background-color: ${({ theme }) => theme.colors.secondary[0]};

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
`
