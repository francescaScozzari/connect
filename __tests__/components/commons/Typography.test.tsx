import { expect } from '@jest/globals'
import { screen } from '@testing-library/react'
import React from 'react'

import { renderWithWrappers } from '@/__tests__/functions'
import { H1, H2, H3, H4, Text } from '@/components/commons/Typography'

describe('Titles', () => {
  describe('H1', () => {
    it('Normal H1 renders correctly', () => {
      renderWithWrappers(<H1.Normal>MainTitle</H1.Normal>)
      screen.getByText('MainTitle')
    })
    it('Normal H1 renders correctly with style props', () => {
      renderWithWrappers(
        <H1.Normal uppercase italic color='red'>
          MainTitle
        </H1.Normal>
      )
      expect(screen.getByText('MainTitle')).toMatchSnapshot()
    })
    it('Big H1 renders correctly', () => {
      renderWithWrappers(<H1.Big>MainTitle</H1.Big>)
      screen.getByText('MainTitle')
    })
    it('Big H1 renders correctly with style props', () => {
      renderWithWrappers(
        <H1.Big uppercase italic color='red'>
          MainTitle
        </H1.Big>
      )
      expect(screen.getByText('MainTitle')).toMatchSnapshot()
    })
    it('Medium H1 renders correctly', () => {
      renderWithWrappers(<H1.Medium>MainTitle</H1.Medium>)
      screen.getByText('MainTitle')
    })
    it('Medium H1 renders correctly with style props', () => {
      renderWithWrappers(
        <H1.Medium uppercase italic color='red'>
          MainTitle
        </H1.Medium>
      )
      expect(screen.getByText('MainTitle')).toMatchSnapshot()
    })
  })
  describe('H2', () => {
    it('Normal H2 renders correctly', () => {
      renderWithWrappers(<H2.Normal>MainTitle</H2.Normal>)
      screen.getByText('MainTitle')
    })
    it('Normal H2 renders correctly with style props', () => {
      renderWithWrappers(
        <H2.Normal uppercase italic color='red'>
          MainTitle
        </H2.Normal>
      )
      expect(screen.getByText('MainTitle')).toMatchSnapshot()
    })
  })
  describe('H3', () => {
    it('H3 renders correctly', () => {
      renderWithWrappers(<H3.Normal>MainTitle</H3.Normal>)
      screen.getByText('MainTitle')
    })
    it('H3 renders correctly with style props', () => {
      renderWithWrappers(
        <H3.Normal uppercase italic color='red'>
          MainTitle
        </H3.Normal>
      )
      expect(screen.getByText('MainTitle')).toMatchSnapshot()
    })
  })
  describe('H4', () => {
    it('H4 renders correctly', () => {
      renderWithWrappers(<H4.Normal>MainTitle</H4.Normal>)
      screen.getByText('MainTitle')
    })
    it('H4 renders correctly with style props', () => {
      renderWithWrappers(
        <H4.Normal uppercase italic color='red'>
          MainTitle
        </H4.Normal>
      )
      expect(screen.getByText('MainTitle')).toMatchSnapshot()
    })
  })
})

describe('Text', () => {
  describe('Normal', () => {
    it('Normal Text renders correctly', () => {
      renderWithWrappers(<Text.Normal>MainParagraph</Text.Normal>)
      screen.getByText('MainParagraph')
    })
    it('Normal Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.Normal uppercase italic color='red'>
          MainParagraph
        </Text.Normal>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
    it('NormalBold Text renders correctly', () => {
      renderWithWrappers(<Text.NormalBold>MainParagraph</Text.NormalBold>)
      screen.getByText('MainParagraph')
    })
    it('NormalBold Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.NormalBold uppercase italic color='red'>
          MainParagraph
        </Text.NormalBold>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
  })
  describe('Small', () => {
    it('Small Text renders correctly', () => {
      renderWithWrappers(<Text.Small>MainParagraph</Text.Small>)
      screen.getByText('MainParagraph')
    })
    it('Small Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.Small uppercase italic color='red'>
          MainParagraph
        </Text.Small>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
    it('SmallBold Text renders correctly', () => {
      renderWithWrappers(<Text.SmallBold>MainParagraph</Text.SmallBold>)
      screen.getByText('MainParagraph')
    })
    it('SmallBold Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.SmallBold uppercase italic color='red'>
          MainParagraph
        </Text.SmallBold>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
  })
  describe('Medium', () => {
    it('Medium Text renders correctly', () => {
      renderWithWrappers(<Text.Medium>MainParagraph</Text.Medium>)
      screen.getByText('MainParagraph')
    })
    it('Medium Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.Medium uppercase italic color='red'>
          MainParagraph
        </Text.Medium>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
    it('MediumBold Text renders correctly', () => {
      renderWithWrappers(<Text.MediumBold>MainParagraph</Text.MediumBold>)
      screen.getByText('MainParagraph')
    })
    it('MediumBold Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.MediumBold uppercase italic color='red'>
          MainParagraph
        </Text.MediumBold>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
  })
  describe('Big', () => {
    it('Big Text renders correctly', () => {
      renderWithWrappers(<Text.Big>MainParagraph</Text.Big>)
      screen.getByText('MainParagraph')
    })
    it('Big Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.Big uppercase italic color='red'>
          MainParagraph
        </Text.Big>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
    it('BigBold Text renders correctly', () => {
      renderWithWrappers(<Text.BigBold>MainParagraph</Text.BigBold>)
      screen.getByText('MainParagraph')
    })
    it('BigBold Text renders correctly with style props', () => {
      renderWithWrappers(
        <Text.BigBold uppercase italic color='red'>
          MainParagraph
        </Text.BigBold>
      )
      expect(screen.getByText('MainParagraph')).toMatchSnapshot()
    })
  })
})
