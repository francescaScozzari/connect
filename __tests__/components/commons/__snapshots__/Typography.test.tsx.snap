// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Big Big Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 18px;
  font-weight: 400;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Big BigBold Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 18px;
  font-weight: 700;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Medium Medium Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 12px;
  font-weight: 400;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Medium MediumBold Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 12px;
  font-weight: 700;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Normal Normal Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 16px;
  font-weight: 400;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Normal NormalBold Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 16px;
  font-weight: 700;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Small Small Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 10px;
  font-weight: 400;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Text Small SmallBold Text renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 10px;
  font-weight: 500;
}

<p
  class="c0 c1"
  color="red"
>
  MainParagraph
</p>
`;

exports[`Titles H1 Big H1 renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 50px;
  font-weight: 500;
}

<h1
  class="c0 c1"
  color="red"
>
  MainTitle
</h1>
`;

exports[`Titles H1 Medium H1 renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 45px;
  font-weight: 500;
}

<h1
  class="c0 c1"
  color="red"
>
  MainTitle
</h1>
`;

exports[`Titles H1 Normal H1 renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 40px;
  font-weight: 500;
}

<h1
  class="c0 c1"
  color="red"
>
  MainTitle
</h1>
`;

exports[`Titles H2 Normal H2 renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 32px;
  font-weight: 500;
}

<h2
  class="c0 c1"
  color="red"
>
  MainTitle
</h2>
`;

exports[`Titles H3 H3 renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 24px;
  font-weight: 500;
}

<h3
  class="c0 c1"
  color="red"
>
  MainTitle
</h3>
`;

exports[`Titles H4 H4 renders correctly with style props 1`] = `
.c0 {
  color: red;
  margin: 0;
  text-transform: uppercase;
  font-style: italic;
}

.c1 {
  font-size: 20px;
  font-weight: 500;
}

<h4
  class="c0 c1"
  color="red"
>
  MainTitle
</h4>
`;
